src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,416 @@
+package com.phonebookservice.util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class MyArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] data;
+    private int index = 0;
+
+    /**
+     * Initialization of array.
+     */
+    public MyArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    /**
+     * Initialization of array.
+     *
+     * @param size the size.
+     */
+    public MyArrayList(final int size) {
+        super();
+        if (size < 0) {
+            throw new IllegalArgumentException("Invalid size: " + size
+                    + ". The size should be positive number");
+        }
+        this.data = (T[]) new Object[size];
+    }
+
+    /**
+     * Appends the element to the end of this list.
+     *
+     * @param element the element.
+     * @return true if added else false.
+     */
+    @Override
+    public boolean add(final T element) {
+        if (this.index > this.data.length - 1) {
+            this.increaseSize();
+        }
+
+        this.data[this.index] = element;
+        this.index++;
+        return true;
+    }
+
+    /**
+     * Inserts the element at the specified position in this list.
+     *
+     * @param index   the index.
+     * @param element the element.
+     */
+    @Override
+    public void add(final int index, final T element) {
+        this.checkIndex(index);
+
+        if (index <= this.index) {
+            if (this.index > this.data.length - 1) {
```

Comment:
U increase the size when 
This.index == this.data.Index 

If this.index > this.dat.index you will throw index out of bound in check method